torch/nn/modules/module.py

```diff
@@ -2005,6 +2022,11 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
             exists in :attr:`state_dict`, :meth:`load_state_dict` will raise a
             ``RuntimeError``.
         """
+        if assign:
```

Comment:
@jbschlosser the case you mentioned where we accidentally do 

```
with torch.device('meta'):
    m = Module()
opt = torch.optim.SGD(m.params())
m.load_state_dict(state_dict, assign=True)
out = net(inp)
out.sum().backward()
opt.step()
opt.zero_grad()
```

doesn't seem to error, it seems like [all the p.grad here](url) will be None so it would just be silently incorrect, so decided to warn here instead